src/views/Wrap/index.tsx

```diff
@@ -0,0 +1,331 @@
+    },
+    '& .MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline': {
+      borderColor: theme.palette.mode.lightGray300,
+    },
+    '& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline': {
+      borderColor: theme.palette.mode.lightGray300,
+    },
+    '& .MuiOutlinedInput-notchedOutline': {
+      borderColor: theme.palette.mode.lightGray300,
+    },
+  },
+}));
+
+function a11yProps(index: number) {
+  return {
+    id: `simple-tab-${index}`,
+    'aria-controls': `simple-tabpanel-${index}`,
+  };
+}
+
+function Wrap() {
+  const { t } = useTranslation();
+  const styles = useStyles();
+  const dispatch = useDispatch();
+  const { provider, address, connect, chainID } = useWeb3Context();
+  const tabsActions = useRef<TabsActions>(null);
+  const currenTheme = useContext(AppThemeContext).name;
+  const isSmallScreen = useMediaQuery('(max-width: 600px)');
+
+  const [view, setView] = useState(0);
+  const [quantity, setQuantity] = useState<string>('');
+
+  const [open, setOpen] = useState(false);
+  const [action, setAction] = useState<string>('');
+
+  const [indexAdjustedsClam, setIndexAdjustedsClam] = useState(0);
+  const [predictedPearl, setPredictedPearl] = useState(0);
+  const [predictedsClam, setPredictedsClam] = useState(0);
+
+  const isAppLoading = useSelector<IReduxState, boolean>(state => state.app.loading);
+  const currentIndex = useSelector<IReduxState, string>(state => {
+    return state.app.currentIndex;
+  });
+  const pearlBalance = useSelector<IReduxState, string>(state => state.account.balances?.pearl);
+  const sClamBalance = useSelector<IReduxState, string>(state => state.account.balances?.sClam);
+
+  const wrapAllowance = useSelector<IReduxState, number>(state => state.account.wrapping?.sClamWrap);
+  const pendingTransactions = useSelector<IReduxState, IPendingTxn[]>(state => state.pendingTransactions);
+
+  const isWrapTab = () => view === 0;
+
+  const wrapImage = () => {
+    if (isWrapTab() && currenTheme === 'light') return WrapImageLight;
+    if (isWrapTab() && currenTheme === 'dark') return WrapImageDark;
+    if (!isWrapTab() && currenTheme === 'light') return UnwrapImageLight;
+    if (!isWrapTab() && currenTheme === 'dark') return UnwrapImageDark;
+  };
+
+  const setMax = () => {
+    if (isWrapTab()) {
+      setQuantity(sClamBalance);
+    } else {
+      setQuantity(pearlBalance);
+    }
+  };
+
+  const handleOpenDialog = () => {
+    setOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setOpen(false);
+    setQuantity('');
+  };
+
+  const onApproval = async () => {
+    await dispatch(approveWrapping({ address, provider, networkID: chainID }));
+  };
+
+  const onChangeWrap = async (action: string) => {
+    if (isNaN(Number(quantity)) || Number(quantity) === 0 || quantity === '0') {
+      alert('Please enter a value!');
+    } else {
+      setAction(action);
+      let wrapTx: any = await dispatch(
```

Comment:
Try not to use `any`

Reply:
https://github.com/OtterClam/otter-frontend/pull/27/commits/864af78ceb9dbcc8f7239ebbd56095d657f90a3f

Reply:
reference: https://redux-toolkit.js.org/tutorials/typescript